test/server/graphql/v2/mutation/MemberInvitationMutations.test.ts

```diff
@@ -0,0 +1,413 @@
+import { expect } from 'chai';
+import gqlV2 from 'fake-tag';
+import { describe, it } from 'mocha';
+import { createSandbox } from 'sinon';
+
+import ActivityTypes from '../../../../../server/constants/activities';
+import roles from '../../../../../server/constants/roles';
+import { idEncode, IDENTIFIER_TYPES } from '../../../../../server/graphql/v2/identifiers';
+import emailLib from '../../../../../server/lib/email';
+import models from '../../../../../server/models';
+import * as utils from '../../../../utils';
+
+let host, user1, user2, user3, user4, user5, user6, collective1, collective2;
+let createdMemberInvitationId1, createdMemberInvitationId2, createdMemberInvitationId3;
+let sandbox, sendEmailSpy;
+
+describe('MemberInvitationMutations', () => {
+  /* SETUP
+    - collective1: host, user1 as admin
+    - user2, user3, ... user6
+    - collective2: user2 as admin
+  */
+  before(() => {
+    sandbox = createSandbox();
+    sendEmailSpy = sandbox.spy(emailLib, 'sendMessage');
+  });
+
+  after(() => sandbox.restore());
+  before(() => utils.resetTestDB());
+  before(async () => {
+    user1 = await models.User.createUserWithCollective(utils.data('user1'));
+  });
+  before(async () => {
+    host = await models.User.createUserWithCollective(utils.data('host1'));
+  });
+  before(async () => {
+    user2 = await models.User.createUserWithCollective(utils.data('user2'));
+  });
+  before(async () => {
+    user3 = await models.User.createUserWithCollective(utils.data('user3'));
+  });
+  before(async () => {
+    user4 = await models.User.createUserWithCollective(utils.data('user4'));
+  });
+  before(async () => {
+    user5 = await models.User.createUserWithCollective(utils.data('user5'));
+  });
+  before(async () => {
+    user6 = await models.User.createUserWithCollective(utils.data('user6'));
+  });
+  before(async () => {
+    collective1 = await models.Collective.create(utils.data('collective1'));
+  });
+  before(async () => {
+    collective2 = await models.Collective.create(utils.data('collective2'));
+  });
+  before(() => collective1.addUserWithRole(host, roles.HOST));
+  before(() => collective1.addUserWithRole(user1, roles.ADMIN));
+  before(() => collective2.addUserWithRole(host, roles.HOST));
+  before(() => collective2.addUserWithRole(user2, roles.ADMIN));
```

Comment:
That's very verbose. The latest best practice is to use the helpers from `fake-data`. We can rewrite as something like:

```suggestion
  before(async () => {
    await utils.resetTestDB();
    sandbox = createSandbox();
    sendEmailSpy = sandbox.spy(emailLib, 'sendMessage');
    user = await fakeUser();
    collective = await fakeCollective({ admin: user });
    host = collective.host;
  });
```

I also don't get why we need so many users? If they have different settings, could we make that clearer by using more explicit variable names? Example:

```es6
collectiveAdminUser = await fakeUser();
hostAdminUser = await fakeUser();
```

Reply:
Yeah, they have different settings. I've used better naming now and migrated to the `fakeUser()` utility.